src/prefect/cache_policies.py

```diff
@@ -1,17 +1,33 @@
 import inspect
 from dataclasses import dataclass, field
-from typing import Any, Callable, Dict, List, Optional
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Literal, Optional, Union
+
+from typing_extensions import Self
 
 from prefect.context import TaskRunContext
 from prefect.utilities.hashing import hash_objects
 
+if TYPE_CHECKING:
+    from prefect.filesystems import WritableFileSystem
+    from prefect.locking.protocol import LockManager
+    from prefect.transactions import IsolationLevel
+
 
 @dataclass
 class CachePolicy:
     """
     Base class for all cache policies.
     """
 
+    storage: Union["WritableFileSystem", str, Path, None] = None
```

Comment:
Only because now is the time to ask these questions - are there any other name options we have for this one, just to make crystal clear it's not the same thing as result storage?

Reply:
Another possibility could be `cache_storage`. It'd be a tad redundant, but it'd be explicit.

Reply:
A few other ideas just to explore our options:
- `key_storage`
- `metadata_storage` (mirrors the internal API)
- `key_files`
- `metadata_files`
- `cache_files`
- `cache_data`
- `metadata`